workers/unity/Packages/com.improbable.gdk.transformsynchronization/Systems/DirectTransformUpdateSystem.cs

```diff
@@ -19,26 +15,48 @@ namespace Improbable.Gdk.TransformSynchronization
     [UpdateBefore(typeof(DefaultUpdateLatestTransformSystem))]
     public class DirectTransformUpdateSystem : ComponentSystem
     {
-        private struct Data
+        private WorkerSystem worker;
+        private ComponentUpdateSystem updateSystem;
+
+        private ComponentGroup transformGroup;
+
+        protected override void OnCreateManager()
         {
-            public readonly int Length;
-            [WriteOnly] public ComponentDataArray<TransformToSet> TransfomToSet;
-            [ReadOnly] public ComponentDataArray<TransformInternal.Component> Transform;
+            base.OnCreateManager();
 
-            [ReadOnly] public ComponentDataArray<TransformInternal.ReceivedUpdates> DenotesReceivedUpdate;
-            [ReadOnly] public ComponentDataArray<NotAuthoritative<TransformInternal.Component>> DenotesNotAuthoritative;
-            [ReadOnly] public ComponentDataArray<DirectReceiveTag> DenotesShouldUseDirectReceive;
-        }
+            worker = World.GetExistingManager<WorkerSystem>();
+            updateSystem = World.GetExistingManager<ComponentUpdateSystem>();
 
-        [Inject] private Data data;
-        [Inject] private WorkerSystem worker;
+            transformGroup = GetComponentGroup(
+                ComponentType.Create<TransformToSet>(),
+                ComponentType.ReadOnly<TransformInternal.Component>(),
+                ComponentType.ReadOnly<SpatialEntityId>(),
+                ComponentType.ReadOnly<DirectReceiveTag>(),
+                ComponentType.ReadOnly<TransformInternal.ComponentAuthority>());
+        }
 
         protected override void OnUpdate()
         {
-            for (int i = 0; i < data.Length; ++i)
+            transformGroup.SetFilter(TransformInternal.ComponentAuthority.NotAuthoritative);
+
+            var transformToSetArray = transformGroup.GetComponentDataArray<TransformToSet>();
+            var spatialEntityIdArray = transformGroup.GetComponentDataArray<SpatialEntityId>();
+            var transformComponentArray = transformGroup.GetComponentDataArray<TransformInternal.Component>();
+
+            for (int i = 0; i < transformToSetArray.Length; ++i)
             {
-                var t = data.Transform[i];
-                data.TransfomToSet[i] = new TransformToSet
+                var entityId = spatialEntityIdArray[i];
+                // todo this means an api is missing
```

Comment:
What API is missing?

Reply:
Which entities have updates on them. Ultimately I think transforms might just need their own semi-reactive component.

Reply:
Will remove the comment though.